src/services/stringCompletions.ts

```diff
@@ -448,10 +448,27 @@ namespace ts.Completions.StringCompletions {
 
         fragment = ensureTrailingDirectorySeparator(fragment);
 
-        // const absolutePath = normalizeAndPreserveTrailingSlash(isRootedDiskPath(fragment) ? fragment : combinePaths(scriptPath, fragment)); // TODO(rbuckton): should use resolvePaths
         const absolutePath = resolvePath(scriptPath, fragment);
         const baseDirectory = hasTrailingDirectorySeparator(absolutePath) ? absolutePath : getDirectoryPath(absolutePath);
 
+        // check for a version redirect
+        const packageJsonPath = findPackageJson(baseDirectory, host);
+        if (packageJsonPath) {
+            const packageJson = readJson(packageJsonPath, host as { readFile: (filename: string) => string | undefined });

```

Comment:
What's the cast needed for?

Reply:
I don’t know (this code just moved), but I’m not investigating this one right now because when I fix the same issue but for package.json `exports` I’m going to migrate this to use the PackageJsonCache. Did you know that `readJson` actually creates a JSON `SourceFile` object and fills it with parse diagnostics? 🙃 This is pretty much the least efficient way we could possibly get the contents of a package.json as an object.